ios/packages/test-utils/Sources/ui-test/SegmentControlView.swift

```diff
@@ -0,0 +1,231 @@
+        pluginSections: [FlowLoader.FlowSection],
+        padding: CGFloat = 16
+    ) {
+        self.plugins = plugins
+        self.assetSections = assetSections
+        self.pluginSections = pluginSections
+        self.padding = padding
+    }
+
+    enum HeaderSelection: String, CaseIterable {
+        case flows = "Asset Flows"
+        case player = "Plugins + Managed Player"
+    }
+
+    @State var segmentationSelection: HeaderSelection = .flows
+
+    @State var pubsubEventPublished = false
+    @State var pubsubEventName = ""
+    @State var pubsubEventMessage = ""
+
+    @State var beaconsRecieved = false
+    @State var beaconsInfo = ""
+
+    @State var doneFlow = false
+    @State var outcome = ""
+
+    public var body: some View {
+        VStack {
+            Picker("", selection: $segmentationSelection) {
+                ForEach(HeaderSelection.allCases, id: \.self) { option in
+                    Text(option.rawValue)
+                }
+            }.pickerStyle(SegmentedPickerStyle())
+
+            Spacer()
+
+            if segmentationSelection == .flows {
+                flowsListSection
+            } else {
+                playerListSection
+            }
+        }
+    }
+
+    func assetFlowCompletion(result: Result<CompletedState, PlayerError>) {
+        doneFlow = true
+        switch result {
+        case .success(let result):
+            outcome = result.endState?.outcome ?? "No Outcome"
+        case .failure(let error):
+            guard case let .promiseRejected(errorState) = error else {
+                outcome = error.playerDescription
+                return
+            }
+
+            outcome = errorState.error
+        }
+    }
+
+    var flowsListSection: some View {
+        return AssetCollection(
+            plugins: plugins,
+            sections: assetSections,
+            completion: self.assetFlowCompletion(result:)
+        )
+        .accessibility(identifier: "AssetCollection")
+        .navigationBarTitle(Text("Flows"))
+        .alert(isPresented: $doneFlow, content: {
+            Alert(title: Text("FlowFinished"),
+                    message: Text("Outcome: \(outcome)"),
+                    dismissButton: .default(Text("OK")))
+        })
+    }
+
+    var playerListSection: some View {
+        List {
+            // Plugin flows
+            ForEach(pluginSections, id: \.title) { section in
+                Section {
+                    ForEach(section.flows, id: \.name) { flow in
+                        NavigationLink(flow.name) {
+                            PlayerLoader(flow: flow.flow)
+                                .navigationBarTitle(Text(flow.name))
+                        }
+                        .accessibility(identifier: "\(section.title) \(flow.name)")
+                    }
+                } header: {
+                    Text(section.title)
+                }
+            }
+
+            // Managed Player flows
+            Section {
```

Comment:
Since this is tightly coupled to  ManagedPlayer example flows, we should probably move this file to the demo app, so as not to publish it for users that consume the UI test utilities